### PR TITLE
[concepts]: Out of Memory Conditions

### PR DESCRIPTION
**Pull request by brendan-stephens**
# What changed, and why it matters
Rework: service-memory-limits to move the list of services
into an include file. This allows for multiple articles to reference the
same set with only 1 file to maintain.

Added: out-of-memory conditions to explain OOM behaviour
and how Aiven uses containers to prevent uncontrolled shutdowns.
This is also a replacement for:
https://help.aiven.io/en/articles/4509218-out-of-memory-killer

### CODE REVIEW

File: docs/platform/concepts/out-of-memory-conditions.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+Which process will be killed?
+-----------------------------
+
+The ``OOM Killer`` selects process to kill based on a ``oom_score``; a calculation that balances the how much memory the process is using with how long the process has been running. 
+
+Processes that have been running for a long time are less likely to be killed. Subprocesses are summed with parent processes in terms of memory usage, so a process which forks a large number of subprocesses, but itself does not use a lot of memory, may still be killed.
+
+*In most instances, the hosted data service, or a child process, will have the highest memory footprint and be a prime candidate for termination when the OOM Killer sweeps the running processes.*
+
+Aiven's cloud data platform leverages kernel namespaces (or containers) to isolate processes from each other. Isolation has several benefits, including:
```
Comment (TibsAtWork) [inline docs/platform/concepts/out-of-memory-conditions.rst:28]: Is `contsainers` here another name for `kernel namespaces`? Is so, we can drop the `or` in front of the word `containers`

**Comment by TibsAtWork** [comment]
This looks good to me - are the numbers in still correct?

File: docs/platform/concepts/out-of-memory-conditions.rst (hunk 1 above)
Comment (brendan-stephens) [reply docs/platform/concepts/out-of-memory-conditions.rst:28]: The word “container” doesn’t mean anything super precise. Basically there are a few new Linux kernel features (“namespaces” and “cgroups”) that let you isolate processes from each other. When you use those features, you call it “containers”.

**Comment by brendan-stephens** [comment]
There are several articles that reference overhead, each one providing a calculation, and if we change the overhead value (350 MB) to something else, there would end up being multiple documents to manage. 

The numbers are current, but in my next PR, I might move the calculations to an include, but I have to think of a good way to do that. It might mean moving from a math block to a code block.

File: docs/platform/concepts/out-of-memory-conditions.rst (hunk 1 above)
Comment (TibsAtWork) [reply docs/platform/concepts/out-of-memory-conditions.rst:28]: Thanks - I thought I'd check. In that case the text is perfectly OK as it is.

**Review by TibsAtWork — APPROVED** [review]
Thanks very much for doing this.
Looks nice and clear to me.

**Pull request merged**